maltoolbox/attackgraph/attackgraph.py

```diff
@@ -332,8 +172,8 @@ def _from_dict(
                 'is_necessary' in node_dict else True
             ag_node.mitre_info = str(node_dict['mitre_info']) if \
                 'mitre_info' in node_dict else None
-            ag_node.tags = node_dict['tags'] if \
-                'tags' in node_dict else []
+            ag_node.tags = set(node_dict['tags']) if \
+                'tags' in node_dict else set()
```

Comment:
```suggestion
            ag_node.tags = set(node_dict.get('tags', []))
```

Would it be better to return a set from the compiler already? I think it would. In general, there are such checks scattered all over the place in the app code and we could keep it less noisy if the compiler did such things. I have a TA that started working today on the compiler. I can ask him to review the toolbox code for such things and make a list of things that we already do in the compiler.

Reply:
This is in `from_dict` I notice now, so it is an invalid comment.